dts/bindings/serial/adi,max32-uart.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: MAX32 UART
+
+include: [uart-controller.yaml, pinctrl-device.yaml]
+
+compatible: "adi,max32-uart"
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+
+  interrupts:
+    required: true
+
+  pinctrl:
+    type: phandles
+
+  clock-source:
+    type: int
+    enum: [0, 1, 2, 3, 4, 5]
+    description: |
+      Clock source to be used by the UART peripheral. The following options
+      are available:
+      - 0: "ADI_MAX32_PRPH_CLK_SRC_PCLK" Peripheral clock
+      - 1: "ADI_MAX32_PRPH_CLK_SRC_EXTCLK" External Clock
+      - 2: "ADI_MAX32_PRPH_CLK_SRC_IBRO" Internal Baud Rate Oscillator
+      - 3: "ADI_MAX32_PRPH_CLK_SRC_ERFO" External Radio Frequency Oscillator
+      - 4: "ADI_MAX32_PRPH_CLK_SRC_ERTCO" External Real-Time Clock Oscillator
+      - 5: "ADI_MAX32_PRPH_CLK_SRC_INRO" Internal Ring Oscillator
+      The target device might not support all option please take a look on
+      target device user guide
```

Comment:
provide a header with defines for human readable sources?

Reply:
Thanks,
We have already added [adi_max32_clock.h](https://github.com/zephyrproject-rtos/zephyr/pull/72291/commits/a8ebbccfbf1b67a8bb2e08033e60ccbc30d96a31#diff-6315da564c6dbec63104a705a7164b256e787db64ebe1d24d031053c303de6b3) file which define all possible clock option, and while using we pass this macro to the devicetree node. see: [max32xxx.dtsi#L135](https://github.com/analogdevicesinc/zephyr/blob/add-MAX32690/dts/arm/adi/max32/max32xxx.dtsi#L135)  and [max32690evkit_max32690_m4.dts#L80](https://github.com/analogdevicesinc/zephyr/blob/add-MAX32690/boards/adi/max32690evkit/max32690evkit_max32690_m4.dts#L80)
Multiple peripheral takes clock-source parameters on MAX32 MCUs. Like UART, WDT, Timer... We have already implemented driver for them and other MAX32 partnumbers, all them rely on this use case. so that I would like to keep existing method to eliminate possible side affect.